pyinfra/operations/docker.py

```diff
@@ -0,0 +1,339 @@
+
+    if existent_container and start:
+        if existent_container[0]["State"]["Status"] != "running":
+            yield handle_docker(
+                resource="container",
+                command="start",
+                container=container,
+            )
+
+    if existent_container and not start:
+        if existent_container[0]["State"]["Status"] == "running":
+            yield handle_docker(
+                resource="container",
+                command="stop",
+                container=container,
+            )
+
+    if existent_container and not present:
+        yield handle_docker(
+            resource="container",
+            command="remove",
+            container=container,
+        )
+
+
+@operation(is_idempotent=False)
+def image(image, present=True):
+    """
+    Manage Docker images
+
+    + image: Image and tag ex: nginx:alpine
+    + present: whether the Docker image should be exist
+
+    **Examples:**
+
+    .. code:: python
+
+        # Pull a Docker image
+        docker.image(
+            name="Pull nginx image",
+            image="nginx:alpine",
+            present=True,
+        )
+
+        # Remove a Docker image
+        docker.image(
+            name="Remove nginx image",
+            image:"nginx:image",
+            present=False,
+        )
+    """
+
+    if present:
+        yield handle_docker(
+            resource="image",
+            command="pull",
+            image=image,
+        )
+
+    else:
+        yield handle_docker(
+            resource="image",
+            command="remove",
+            image=image,
+        )
+
+
+@operation()
+def volume(volume, driver="", labels=[], present=True):
+    """
+    Manage Docker volumes
+
+    + volume: Volume name
+    + driver: Docker volume storage driver
+    + labels: Label list to attach in the volume
+    + present: whether the Docker volume should exist
+
+    **Examples:**
+
+    .. code:: python
+
+        # Create a Docker volume
+        docker.volume(
+            name="Create nginx volume",
+            volume="nginx_data",
+            present=True
+        )
+    """
+
+    existent_volume = [v for v in host.get_fact(DockerVolumes) if v["Name"] == volume]
+
+    if present:
+
+        if existent_volume:
+            host.noop("Volume alredy exist!")
+            return
+
+        yield handle_docker(
+            resource="volume",
+            command="create",
+            volume=volume,
+            driver=driver,
+            labels=labels,
+            present=present,
+        )
+
+    else:
+        if existent_volume is None:
+            host.noop("There is no {0} volume!".format(volume))
+            return
+
+        yield handle_docker(
+            resource="volume",
+            command="remove",
+            volume=volume,
+        )
+
+
+@operation()
+def network(
+    network,
+    driver="",
+    gateway="",
+    ip_range="",
+    ipam_driver="",
+    subnet="",
+    scope="",
+    opts=[],
```

Comment:
Default lists here

Reply:
resolved in commit [0ea1976](https://github.com/pyinfra-dev/pyinfra/pull/1114/commits/0ea1976f4fa2881d636e952a0f2018e891d47434)